CodeU_Assignement_2/src/codeu_assignement_2/KemBinaryTree.java

```diff
@@ -0,0 +1,394 @@
+package codeu_assignement_2;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Stack;
+import java.util.Queue;
+import java.util.LinkedList;
+
+/**
+ *
+ * @author Karine
+ * 
+ * Binary tree with a tree node
+ * Each node points to parent, and left/right child
+ * 
+ */
+public class KemBinaryTree <E> {
+    /* Data members of Tree */
+    protected KemTreeNode<E> m_root=null;
+    
+    /* */
+    public KemBinaryTree(){} 
+            
+    /*  Check if the list is empty  */
+    public boolean isEmpty() { return (m_root == null); }
+    
+    /* Allow us to decide where to add items */
+    public KemTreeNode<E> getRoot() { return m_root;}
+    
+    /* Q1: Get all ancestors */
+    public void printAncestors(KemTreeNode<E> child)
+    {  
+        /* Check if child isn't null */
+        if (child == null)
+        {
+            System.out.println("Error: ** Null Pointer **");
+            return;
+        }
+        
+        /* Check if child isn't root */
+        KemTreeNode<E> curr = child.getParent();
+        if (curr == null)
+        {
+            System.out.println("** Root **: has no ancestors. Done.");
+            return;
+        }
+        
+        /* Print all ancestors as long as didn't get to root */
+        while (curr != null)
+        {
+            System.out.print(curr.getData());
+            curr = curr.getParent();
+            if (curr != null)
+            {
+                System.out.print(",");
+            }
+        }
+        System.out.println(". Done printAncestors(..).");
+    }
+
+    /* Q2: Get all ancestors */
+    public KemTreeNode<E> getLowestCommonAncestors(KemTreeNode<E> child1, KemTreeNode<E> child2)
+    {
+        /* No common if at least one is null */
+        if ((child1 == null) || (child2 == null))
+        {
+            return null;
+        }
+        /* If one of them is the root - return it as the common */
+        if ((child1 == m_root) || (child2 == m_root))
+        {
+            return m_root;
+        }
+        /* Couts the edges between the node and the root */
+        int size_c1 = getSizeOfTraceChild2Root(child1);
+        int size_c2 = getSizeOfTraceChild2Root(child2);
+        if ((size_c1 == -1) || (size_c2 == -1))

```

Comment:
To mark it is supposed to be unsigned. I tried: @Unsigned before the method. Maybe it is better?